addons/account/models/account_move_line.py

```diff
@@ -691,7 +691,10 @@ def _compute_currency_rate(self):
 
     def _get_rate_date(self):
         self.ensure_one()
-        return self.move_id.invoice_date or self.move_id.date or fields.Date.context_today(self)
+        if self.move_id.move_type in ('in_refund', 'out_refund'):
+            return self.move_id.date or self.move_id.invoice_date or fields.Date.context_today(self)
```

Comment:
Shouldn't we look at `move_id.reversed_entry_id._get_rate_date()` instead?

Also, why is the invoice date not the same as the one from the invoice if you need that same rate?

Note that in Odoo 18 you can chose the rate by setting `invoice_currency_rate` https://github.com/odoo/odoo/commit/2d486fb93ba6d06821837a8f635b8efe44dde00e